src/OpenTracing/ISpan.cs

```diff
@@ -1,164 +1,133 @@
+        /// span.Log(
+        /// new Dictionary{string, object}
+        /// {
+        ///   { "event", "soft error" },
+        ///   { "type", "cache timeout" },
+        ///   { "waited.millis", 1500 },
         /// });
         /// </code>
-        /// </example>
-        ISpan Log(IEnumerable<KeyValuePair<string, object>> fields);
+        /// </para>
+        /// </summary>
+        /// <param name="fields">
+        /// key:value log fields. Tracer implementations should support String, numeric, and boolean values;
+        /// some may also support arbitrary objects.
+        /// </param>
+        /// <returns>The Span, for chaining</returns>
+        /// <seealso cref="Log(string)"/>
+        ISpan Log(IDictionary<string, object> fields);
 
         /// <summary>
-        /// <para>Log key:value pairs to the span with an explicit timestamp.</para>
-        /// <para>CAUTIONARY NOTE: Not all Tracer implementations support key:value log fields end-to-end.
-        /// It is possible to pass a list of key:value pairs instead of a dictionary. However, the behavior
-        /// for lists with duplicate keys is undefined.
-        /// Caveat emptor.</para>
+        /// Like <see cref="Log(IDictionary{string, object})"/>, but with an explicit timestamp.
+        /// <para><em>CAUTIONARY NOTE:</em> not all Tracer implementations support key:value log fields end-to-end. Caveat emptor.</para>
         /// </summary>
-        /// <param name="timestamp">The explicit timestamp for the log record. Must be greater than or equal to the
-        /// span's start timestamp.</param>
+        /// <param name="timestampMicroseconds">
+        /// The explicit timestamp for the log record. Must be greater than or equal to the Span's start
+        /// timestamp.
+        /// </param>
         /// <param name="fields">
-        ///   <para>key:value log fields. Tracer implementations should support string, numeric, and boolean values;
-        ///   some may also support arbitrary objects.</para>
-        ///   <para>The behavior for lists with duplicate keys is undefined.</para>
+        /// key:value log fields. Tracer implementations should support String, numeric, and boolean values;
+        /// some may also support arbitrary objects.
         /// </param>
-        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
-        /// <example>
-        /// <code>
-        /// span.Log(new Dictionary&lt;string, object&gt; {
-        ///     { "event", "soft error" },
-        ///     { "type", "cache timeout" },
-        ///     { "waited.millis", 1500 }
-        /// });
-        /// </code>
-        /// </example>
-        ISpan Log(DateTimeOffset timestamp, IEnumerable<KeyValuePair<string, object>> fields);
+        /// <returns>The Span, for chaining</returns>
+        /// <seealso cref="Log(long, string)"/>
+        ISpan Log(long timestampMicroseconds, IDictionary<string, object> fields);
```

Comment:
Please keep the usage of DateTimeOffset everywhere. That difference was discussed in #4 and #42 and makes a lot of sense for .NET.

Reply:
Will change but want to bring onto your radar my logic here:

So I was even involved in one of those I think, but in the past few months have been focused on perf at work. I have a minor concern regarding it here as DateTimeOffset is less performant than a long (e.g. to make the DateTimeOffset the caller has to parse from the clock into this struct anyway, and then if the tracing system does NOT want to use that first hand - which is quite possible given the java APIs expect long which is more cross-language - they'd have to parse it back to a long, a round trip that isn't strictly necessary.

That said, we can cross this bridge when we come to it (e.g. after we have some real implementations that have all other perf issues fixed, as it's unlikely this will be the bottleneck in any implementation at first)

Reply:
There isn't even a way to get the ticks without going through a DateTime(Offset), is there?